### PR TITLE
Add a command to execute the previous test

### PR DESCRIPTION
**Pull request by ironcladlou**
Add a `go.test.previous` command which runs the last test which was
executed.

### CODE REVIEW

**Comment by ironcladlou** [comment]
@lukehoban do you have any idea what's up with those Travis failures?

**Comment by ramya-rao-a** [comment]
@ironcladlou The test for gometalinter tries to set the user settings for `linttool`. 
```
let config = vscode.workspace.getConfiguration('go');
config['lintTool'] = 'gometalinter';
```
which fails with the latest version of VSCode as you can no longer update config that way.

There have been changes to the configuration service https://github.com/Microsoft/vscode/issues/1396
I'll look into updating the test case

File: src/goMain.ts [hunk 1]
```diff
@@ -72,6 +72,11 @@ export function activate(ctx: vscode.ExtensionContext): void {
 		testCurrentFile(goConfig['testTimeout']);
 	}));
 
+	ctx.subscriptions.push(vscode.commands.registerCommand('go.test.previous', () => {
+		let goConfig = vscode.workspace.getConfiguration('go');
+		testPrevious();
```
Comment (lukehoban) [inline src/goMain.ts]: Do you intend to pass `testTimeout` here as in other test commands?  Or to not use `goConfig`?

File: src/goTest.ts [hunk 2]
```diff
@@ -112,12 +116,32 @@ export function testCurrentFile(timeout: string) {
 }
 
 /**
+ * Runs the previously executed test.
+ */
+export function testPrevious() {
+	let editor = vscode.window.activeTextEditor;
+	if (!editor) {
```
Comment (lukehoban) [inline src/goTest.ts]: Why does an editor need to be active in this case?  It sounds like the semantics for this command are not tied to what is in the current editor (if there is one).  It just executes whatever the last Go test was that was invoked.

File: src/goMain.ts (hunk 1 above)
Comment (lukehoban) [inline src/goMain.ts]: Actually - looks like you are intentionally not passing `testTimeout` - so should drop the `goConfig` line.

File: src/goTest.ts (hunk 2 above)
Comment (ironcladlou) [inline src/goTest.ts]: You're right, I'll remove that check.

File: src/goMain.ts (hunk 1 above)
Comment (ironcladlou) [inline src/goMain.ts]: Removed the config lookup for now.

File: src/goTest.ts (hunk 2 above)
Comment (ironcladlou) [inline src/goTest.ts]: Removed the editor check.

**Comment by ironcladlou** [comment]
@ramya-rao-a, thanks for tracking down that Travis issue!

**Comment by ironcladlou** [comment]
Added a small commit to rename the title of the command to something more auto-complete friendly (as a complement to https://github.com/Microsoft/vscode-go/pull/495).

**Comment by ironcladlou** [comment]
Squashed commits.

**Comment by ironcladlou** [comment]
Will this be merged?

**Comment by ramya-rao-a** [comment]
Sure, I tested your changes and they look good.

Could you trigger another travis runs so that the tests are all green? The last time it failed due to an upstream issue with vscode-debugadapter

**Comment by ironcladlou** [comment]
@ramya-rao-a Thanks! How do I trigger a new build without making some dummy commit?

**Pull request merged**